### PR TITLE
Added level select menu.

### PR DESCRIPTION
**Pull request by jokertyf**

### CODE REVIEW

File: src/states/LevelSelect.gd [hunk 1]
```diff
@@ -0,0 +1,56 @@
+extends CanvasLayer
+
```
Comment (thearst3rd) [inline src/states/LevelSelect.gd:2]: nitpick: add newline

File: src/states/Global.gd [hunk 2]
```diff
@@ -3,6 +3,7 @@ extends Node
 
 const MAX_VOLUME_DB := -6.0
 const SAVE_FILE_PATH := "user://config.cfg"
+const FIRST_LEVEL_PATH = "res://src/levels/IntroLevel.tscn"
```
Comment (thearst3rd) [inline src/states/Global.gd:6]: nitpick: `=` -> `:=`

File: src/states/LevelSelect.gd [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+signal level_select_exited
+
+var level_names := []
+var level_paths := []
+var buttons := []
+
+var page_idx := 0
+
```
Comment (thearst3rd) [inline src/states/LevelSelect.gd:10]: nitpick: add newline

**Review by thearst3rd — CHANGES_REQUESTED** [review]
Once again I'm a pedant so I added some nitpicks but the actual code looks good

**Review by kdreese — APPROVED** [review]

**Review by thearst3rd — APPROVED** [review]

**Pull request merged**